src/test/java/org/elasticsearch/search/aggregations/bucket/TopHitsTests.java

```diff
@@ -265,6 +275,37 @@ public void testBasics() throws Exception {
     }
 
     @Test
+    public void testIssue11119() throws Exception {
+        SearchResponse response = client()
+                .prepareSearch("idx")
+                .setTypes("field-collapsing")
+                .setSize(0)
+                .setQuery(matchQuery("text", "x y z"))
+                .addAggregation(terms("terms").executionHint(randomExecutionHint()).field("group").subAggregation(topHits("hits")))
+                .get();
+
+        assertSearchResponse(response);
```

Comment:
Can we check that the hits information is correct in the response too? i.e. the hits are empty, the totalHits is correct and the maxScore is zero